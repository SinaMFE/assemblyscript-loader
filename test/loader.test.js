
import webpack from './helpers/compiler';

describe('Loader', () => {
  test('Defaults', async () => {
    const config = {
      loader: {
        test: /(ts)/,
        options: {
            name: '[path][name].wasm',
            optimize:"-1",
            validate:true
        }
      },
    };
    const stats = await webpack('fixture.js', config);
    const { source } = stats.toJson().modules[0];
    expect(source).toMatchSnapshot();
  });
});
