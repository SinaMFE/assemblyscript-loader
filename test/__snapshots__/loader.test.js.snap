// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults 1`] = `
"
    function createWebAssemblyModulePromise (deps) {
      var p = new Promise(function(resolve){
        var compatibleModule;
                if (typeof WebAssembly !== 'undefined') {
                    var buffer = new ArrayBuffer(447);
        var uint8 = new Uint8Array(buffer);
        uint8.set([0,97,115,109,1,0,0,0,1,9,2,96,2,127,127,0,96,0,0,3,4,3,0,1,1,4,4,1,112,0,1,5,3,1,0,0,6,16,3,127,1,65,0,11,127,1,65,0,11,127,1,65,0,11,7,32,4,6,109,101,109,111,114,121,2,0,5,116,97,98,108,101,1,0,4,105,110,105,116,0,0,4,115,116,101,112,0,1,9,7,1,0,65,0,11,1,2,10,186,2,3,17,0,32,0,36,0,32,1,36,1,35,0,35,1,108,36,2,11,161,2,1,8,127,35,1,65,1,107,33,6,35,0,65,1,107,33,7,3,64,2,64,32,0,35,1,79,13,0,32,0,65,1,107,32,6,32,0,27,33,4,65,0,32,0,65,1,106,32,0,32,6,70,27,33,5,65,0,33,1,3,64,2,64,32,1,35,0,79,13,0,65,0,32,1,65,1,106,32,1,32,7,70,27,34,2,32,5,35,0,108,106,45,0,0,32,5,35,0,108,32,1,106,45,0,0,32,1,65,1,107,32,7,32,1,27,34,3,32,5,35,0,108,106,45,0,0,32,0,35,0,108,32,2,106,45,0,0,32,0,35,0,108,32,3,106,45,0,0,32,4,35,0,108,32,2,106,45,0,0,32,4,35,0,108,32,3,106,45,0,0,32,4,35,0,108,32,1,106,45,0,0,106,106,106,106,106,106,106,33,2,32,0,35,0,108,32,1,106,45,0,0,4,64,32,2,65,2,72,34,3,4,127,32,3,5,32,2,65,3,74,11,4,64,35,2,32,0,35,0,108,106,32,1,106,65,0,58,0,0,11,5,32,2,65,3,70,4,64,35,2,32,0,35,0,108,106,32,1,106,65,1,58,0,0,11,11,32,1,65,1,106,33,1,12,1,11,11,32,0,65,1,106,33,0,12,1,11,11,11,3,0,1,11,0,31,16,115,111,117,114,99,101,77,97,112,112,105,110,103,85,82,76,13,115,116,101,112,46,119,97,115,109,46,109,97,112]);
        var WebAssemblyModule = function(deps) {
    var defaultDeps = {
        'global': { },
        'env': {
            'memory': new WebAssembly.Memory({
                initial: 10,
                limit: 100}),
            'table': new WebAssembly.Table({
                initial: 0,
                element: 'anyfunc'})
        }
    };
    return new WebAssembly.Instance(new WebAssembly.Module(buffer), deps || defaultDeps);
};

                    compatibleModule = WebAssemblyModule;

                }
                else {
                    \\"use strict\\";
// A simplified version of the game of life as seen on http://dcode.io
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var w, // width
h, // height
s; // total size
/** Initializes width and height. */
function init(w_, h_) {
    w = w_;
    h = h_;
    s = w * h;
}
exports.init = init;
/** Performs one step. */
function step() {
    var hm1 = h - 1, wm1 = w - 1;
    for (var y = 0; y < h; ++y) {
        var ym1 = select(hm1, y - 1, y == 0), yp1 = select(0, y + 1, y == hm1);
        for (var x = 0; x < w; ++x) {
            var xm1 = select(wm1, x - 1, x == 0), xp1 = select(0, x + 1, x == wm1);
            var n = (load(ym1 * w + xm1) + load(ym1 * w + x) + load(ym1 * w + xp1) +
                load(y * w + xm1) + load(y * w + xp1) +
                load(yp1 * w + xm1) + load(yp1 * w + x) + load(yp1 * w + xp1));
            if (load(y * w + x)) {
                if (n < 2 || n > 3)
                    store(s + y * w + x, 0);
            }
            else if (n == 3)
                store(s + y * w + x, 1);
        }
    }
}
exports.step = step;
// Performing a step uses bytes [0, s-1] as the input and writes the output to [s, 2*s-1].
// Note that the code above wastes a lot of space by using one byte per cell.

                    compatibleModule = function() {};          compatibleModule.prototype.exports = exports;
                }
        resolve(compatibleModule(deps).exports);;
      });
      return p
    }
    module.exports = createWebAssemblyModulePromise
    "
`;
